File: Windows Native Runtime/ObjectManager.cpp
#include "ObjectManager.h"

static std::list<GameObject> gameObject_array;

void ObjectManager::loadObject(std::string name, std::string text, std::string img_path, int tag, bool isStatic, bool isPhysics, bool isRigid, bool isCollider, int img_Width, int img_Height)
{
	GameObject instance;

	instance.name = name;
	instance.text = text;

	FILE * file_handle = fopen(img_path.c_str(), "r");

	if (file_handle)
	{
		instance.img.baseImage = IMG_Load(img_path.c_str( ));
		instance.img.Width = img_Width;
		instance.img.Height = img_Height;
	}

	instance.tag = tag;
	instance._static = isStatic;
	instance.physics = isPhysics;
	instance.rigidbody = isRigid;
	instance.collider = isCollider;
	instance.font_name = "Verdana";
	instance.font_size = 12;
	instance.color = { 100, 100, 100, 100};
	
	gameObject_array.push_back(instance);
}

GameObject ObjectManager::findGameObjectWithName(std::string name)
{
	for (std::list<GameObject>::iterator cur_obj = gameObject_array.begin(); cur_obj != gameObject_array.end(); cur_obj++)
	{
		if (cur_obj->name == name)
		{
			return *cur_obj;
		}
	}
}

std::list<GameObject> ObjectManager::findGameObjectWithTag(int tag)
{
	std::list<GameObject> ret_list;

	for (std::list<GameObject>::iterator cur_obj = gameObject_array.begin(); cur_obj != gameObject_array.end(); cur_obj++)
	{
		if (cur_obj->tag == tag)
		{
			 ret_list.push_back (*cur_obj);
		}
	}

	return ret_list;
}

File: Java Desktop Runtime/jruntime/ObjectManager.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package jruntime;

import java.io.IOException;
import java.util.*;
import javax.microedition.lcdui.Image;

/**
 *
 * @author Riju
 */
public class ObjectManager 
{
    static Vector gameObject_array = new Vector( );

        public static void loadObject(String name,String text,String img_path,int tag,boolean isStatic,boolean isPhysics,boolean isRigid,boolean isCollider)
        {
	      GameObject instance = new GameObject( );
		  
		  instance.name = name;
		  instance.text = text;
                  
                  try
                  {  
                        instance.img = Image.createImage(img_path);
                   }
                   catch(IOException ax)
                   {
                    instance.img = null;      
                   }
                   catch(IllegalArgumentException cx)
                   {
                    instance.img = null;   
                   }
                  
		  instance.tag = tag;
		  instance._static = isStatic;
		  instance.physics = isPhysics;
		  instance.rigidbody = isRigid;
		  instance.collider = isCollider;
		  
                  gameObject_array.addElement(instance);
	   }
	   
	   public static Vector findGameObjectWithTag(int tag)
           {
               Vector ret_vec = new Vector( );
               
          for(int cnt = 0;cnt < gameObject_array.size( );cnt++)
          {
		      if (((GameObject) gameObject_array.elementAt(cnt)).tag == tag)
			  {
                              GameObject null_obj = new GameObject( );
                              
                              null_obj.name = ((GameObject) gameObject_array.elementAt(cnt)).name;
                              null_obj.tag = ((GameObject) gameObject_array.elementAt(cnt)).tag;
                              null_obj.text =  ((GameObject)  gameObject_array.elementAt(cnt)).text;
                              null_obj.img = ((GameObject) gameObject_array.elementAt(cnt)).img;
                              null_obj._static = ((GameObject)  gameObject_array.elementAt(cnt))._static;
                              null_obj.collider = ((GameObject) gameObject_array.elementAt(cnt)).collider;
                              null_obj.txt_R  = ((GameObject) gameObject_array.elementAt(cnt)).txt_R;
                              null_obj.txt_G  = ((GameObject) gameObject_array.elementAt(cnt)).txt_G;
                              null_obj.txt_B = ((GameObject) gameObject_array.elementAt(cnt)).txt_B;
                              null_obj.rigidbody = ((GameObject) gameObject_array.elementAt(cnt)).rigidbody;
                              null_obj.physics = ((GameObject) gameObject_array.elementAt(cnt)).physics;
                              null_obj.font = ((GameObject) gameObject_array.elementAt(cnt)).font;
                              
                              ret_vec.addElement(null_obj);		  
			  }
		  }		  
		  
		  return ret_vec;
	   }
	   
	   public static GameObject findGameObjectWithName(String name)
	   {	   
		  for(int cnt = 0;cnt < gameObject_array.size( );cnt++)
          {
		      if (((GameObject) gameObject_array.elementAt(cnt)).name == name)
			  {
                              GameObject null_obj = new GameObject( );
                              
                              null_obj.name = ((GameObject) gameObject_array.elementAt(cnt)).name;
                              null_obj.tag = ((GameObject) gameObject_array.elementAt(cnt)).tag;
                              null_obj.text =  ((GameObject)  gameObject_array.elementAt(cnt)).text;
                              null_obj.img = ((GameObject) gameObject_array.elementAt(cnt)).img;
                              null_obj._static = ((GameObject)  gameObject_array.elementAt(cnt))._static;
                              null_obj.collider = ((GameObject) gameObject_array.elementAt(cnt)).collider;
                              null_obj.txt_R  = ((GameObject) gameObject_array.elementAt(cnt)).txt_R;
                              null_obj.txt_G  = ((GameObject) gameObject_array.elementAt(cnt)).txt_G;
                              null_obj.txt_B = ((GameObject) gameObject_array.elementAt(cnt)).txt_B;
                              null_obj.rigidbody = ((GameObject) gameObject_array.elementAt(cnt)).rigidbody;
                              null_obj.physics = ((GameObject) gameObject_array.elementAt(cnt)).physics;
                              null_obj.font = ((GameObject) gameObject_array.elementAt(cnt)).font;
                              
                  return null_obj;			  
			  }
		  }		  
		  
		  return null;
	   }
}
